dvc/system.py

```diff
@@ -71,8 +71,8 @@ def _reflink_darwin(src, dst):
         clonefile.restype = ctypes.c_int
 
         return clonefile(
-            ctypes.c_char_p(src.encode("utf-8")),
-            ctypes.c_char_p(dst.encode("utf-8")),
+            ctypes.c_char_p(os.fsencode(src)),
+            ctypes.c_char_p(os.fsencode(dst)),
```

Comment:
Ideally, it would be great to have tests that cpython has https://github.com/python/cpython/blob/b5a6db9111562454617b6771b61f2734ea0420c9/Lib/test/support/os_helper.py#L121 for different situations and combinations of undecodable paths.